### ISSUE TITLE
`&` without `?` still creates query params

### ISSUE THREAD

**Issue by veksha** [treeform/webby#19]
fetching this url with puppy fails `https://www.elyrics.net/read/0-9/$tupid-young,-nocap-&-steelz-lyrics/i-can_t-change-lyrics.html`
webby thinks that `&` char is query params delimiter even though `?` char is not present anywhere in URL.

as a workaround i temporarily **commented out** these lines in webby.nim:
```nim
else:
   # Handle ? being ommitted but a search still being present
   let hasAmpersand = s.find('&')
   if hasAmpersand != -1:
     let search = s[hasAmpersand + 1 .. ^1]
     s = s[0 .. hasAmpersand - 1]
     url.query = parseSearch(search)
```

**Comment by guzba** [comment]
The url is malformed, & is a reserved character and must be percent encoded https://www.rfc-editor.org/rfc/rfc3986#section-2.2

There are issues with URL parsing in webby but this is not the correct starting point to fix them.

**Comment by guzba** [comment]
If referenced later, the issue is URL parsing is a separate process from path parsing and webby parseUrl will need a redo.

**Comment by guzba** [comment]
https://github.com/treeform/puppy/issues/111

**Comment by veksha** [comment]
as another starting point we can also try to do breaking down of URL components the same way as regex from rfc https://www.rfc-editor.org/rfc/rfc3986#appendix-B

it doesn't care about `&` char in path:

![image](https://github.com/treeform/webby/assets/275333/b4ee5eb1-8afd-4bc6-9c05-d4889a99b554)

from rfc:
```
it is natural and commonplace to use a regular expression for parsing the
   potential five components of a URI reference.

   The following line is the regular expression for breaking-down a
   well-formed URI reference into its components.
^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?
```

PS: if browsers can get malformed url, why webby/puppy can't?

**Comment by veksha** [comment]
urls also can be tested here: https://www.lookout.net/test/url/url-liveview.html

**Comment by guzba** [comment]
Yes & in path is a case where reserved characters can be used. This should be fixed. I'm not working on this though so I hope someone else can find the time to redo the URL parsing.

**Comment by guzba** [comment]
Work on this was started https://github.com/treeform/webby/pull/18

**Comment by treeform** [comment]
Hey thank you for the issue. I think we have fixed it in the new 0.2.0 version. Feel free to reopen it if still there.
